src/utilities.jl

```diff
@@ -193,19 +232,10 @@ f(x; a = 1, b...) = x
 sig = printmethod(Docs.Binding(Main, :f), f, first(methods(f)))
 ```
 """
-function printmethod(buffer::IOBuffer, binding::Docs.Binding, func, method::Method)
-    # TODO: print qualified?
-    print(buffer, binding.var)
-    print(buffer, "(")
-    join(buffer, arguments(method), ", ")
-    local kws = keywords(func, method)
-    if !isempty(kws)
-        print(buffer, "; ")
-        join(buffer, kws, ", ")
-    end
-    print(buffer, ")")
-    return buffer
-end
+printmethod(buffer::IOBuffer, binding::Docs.Binding, func, method::Method) =
+    printmethod_format(buffer, string(binding.var),
+        string.(arguments(method)),
+        string.(keywords(func, method)))
```

Comment:
Could the broadcast `string` be avoided here and instead just pass through the vectors as is and let `printmethod_format` handle conversion where needed?

Reply:
Not sure -- `printmethod_format` actually only accesses the string-ish version of the thing so I assume that doing the conversion forcibly at one place makes sense here.

Re performance:
- string conversion will need to be done anyway
- here I guess the optimizer has a chance to fuse the `string.` with arguments, avoiding the allocation of the immediate vector altogether. If we converted later, the vector of arguments would have to materialize.

Reply:
Both `arguments` and `keywords` always return `Vector{Symbol}` so probably just pass those through as is and let the final printing as strings happen in `printmethod_format`. Just changing it's signatures to be `Vector{Symbol}`s rather than `Vector{String}` should be sufficient I'd think.

Reply:
ah okay, good, will do

Reply:
ah, that `length` calc for line length will probably have issues with doing `length(::Symbol)`... maybe no way around doing the conversion then.

Reply:
is there any good way to measure the length of a symbol? Seems like I either convert everything right away back to vectors of strings to be able to measure the length, or I do something like `sum(length∘string, args)` and then convert the args to strings again on the end.

Reply:
Probably just go with the current implementation for the time being. If it ever actually became a noticeable performance issue then it can be revisited.

Reply:
Actually, args & type may contain stuff that doesn't really fit nicely into Symbol (see the second call at around L364) -- are you sure it's a good idea to force stuff like spaces into symbols? (Also, the length of a Symbol apparently can't get measured easily -- see julia slack #general a few minutes ago, so I'd basically need to convert it to strings right away anyway.)

Reply:
In that case we might as well adjust those two methods to return `Vector{String}` perhaps instead. That would sidestep most of the issues perhaps.